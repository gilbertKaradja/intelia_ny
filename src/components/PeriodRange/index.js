import React, { Component } from 'react';

class PeriodRange extends Component {

    render() {
        return (
            <div className="PeriodRange">
                {this.props.options.map(item => {
                    return (
                        <div 
                            key={item.value}
                            className={this.props.value.value === item.value ? 'item active': 'item'}
                            onClick={() => { this.props.onChange(item) }}
                        >
                            {item.label}
                        </div>
                    )
                })}
            </div>
        )
    }
}

export default PeriodRange;