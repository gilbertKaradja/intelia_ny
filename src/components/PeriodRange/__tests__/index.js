import React from 'react';
import { shallow, mount } from 'enzyme';

import PeriodRange from '../index.js';

let sampleOptions = [
    { label: 'today', value: '1' },
    { label: 'last week', value: '7' },
    { label: 'last month', value: '30' },
];

describe('PeriodRange component', () => {

    it('should render without issues', () => {

        //renders initial dropdown properly
        const component = shallow(
            <PeriodRange
                value={sampleOptions[0]}
                options={sampleOptions}
            />
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });

    

    it('should call a handler function and reflect value accordingly', () => {

        const onChangeHandler = jest.fn(value => value);

        const component = mount(
            <PeriodRange
                value={sampleOptions[0]}
                options={sampleOptions}
                onChange={onChangeHandler}
            />
        );

        expect(component).toMatchSnapshot();

        component.simulate('click');
        component.children().childAt(1).simulate('click');

        expect(onChangeHandler.mock.calls.length).toBe(1);
        expect(onChangeHandler.mock.results[0].value).toBe(sampleOptions[1]);

        expect(component).toMatchSnapshot();

        component.unmount();

    });
});