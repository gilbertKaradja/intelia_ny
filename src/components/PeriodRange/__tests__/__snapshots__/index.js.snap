// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeriodRange component should call a handler function and reflect value accordingly 1`] = `
<PeriodRange
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "today",
        "value": "1",
      },
      Object {
        "label": "last week",
        "value": "7",
      },
      Object {
        "label": "last month",
        "value": "30",
      },
    ]
  }
  value={
    Object {
      "label": "today",
      "value": "1",
    }
  }
>
  <div
    className="PeriodRange"
  >
    <div
      className="item active"
      key="1"
      onClick={[Function]}
    >
      today
    </div>
    <div
      className="item"
      key="7"
      onClick={[Function]}
    >
      last week
    </div>
    <div
      className="item"
      key="30"
      onClick={[Function]}
    >
      last month
    </div>
  </div>
</PeriodRange>
`;

exports[`PeriodRange component should call a handler function and reflect value accordingly 2`] = `
<PeriodRange
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "label": "last week",
            "value": "7",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Object {
            "label": "last week",
            "value": "7",
          },
        },
      ],
    }
  }
  options={
    Array [
      Object {
        "label": "today",
        "value": "1",
      },
      Object {
        "label": "last week",
        "value": "7",
      },
      Object {
        "label": "last month",
        "value": "30",
      },
    ]
  }
  value={
    Object {
      "label": "today",
      "value": "1",
    }
  }
>
  <div
    className="PeriodRange"
  >
    <div
      className="item active"
      key="1"
      onClick={[Function]}
    >
      today
    </div>
    <div
      className="item"
      key="7"
      onClick={[Function]}
    >
      last week
    </div>
    <div
      className="item"
      key="30"
      onClick={[Function]}
    >
      last month
    </div>
  </div>
</PeriodRange>
`;

exports[`PeriodRange component should render without issues 1`] = `
<div
  className="PeriodRange"
>
  <div
    className="item active"
    key="1"
    onClick={[Function]}
  >
    today
  </div>
  <div
    className="item"
    key="7"
    onClick={[Function]}
  >
    last week
  </div>
  <div
    className="item"
    key="30"
    onClick={[Function]}
  >
    last month
  </div>
</div>
`;
